PE-700 - Add time spent per user report.

Squirrel18 opened the pull request:
## Description:

This PR adds the time spent per user report.
Also adds support for additional request data to be included in the report request and overwrite additional request arguments from the command line.

## Configuration:

This report requires the GOOGLE_OAUTH_CREDENTIALS values.

```
"TIME_SPENT_PER_USER_REPORT": {
    "REPORT_URL": "/proversity-reports/api/v0/generate-time-spent-per-user-report",
    "BACKEND_REPORT": "proversity_reports_script.report_backend.time_spent_per_user_report:TimeSpentPerUserReportBackend",
    "EXTRA_REQUEST_DATA": {
        "date": date to request the report YYYY-MM-DD
    },
    "EXTRA_DATA": {
        "BUCKET_NAME": S3 bucket name,
        "SPREADSHEET_DATA": {
            "time_spent_sheet_id_course-id": Google Spreadsheet id
        }
    }
}
```

## Reviewers:

- [ ] @andrey-canon

andrey-canon commented on proversity_reports_script/report_backend/time_spent_per_user_report.py:
```diff
@@ -0,0 +1,292 @@
+    return report_course_data
+
+
+def grouping_report_data_by_key(course_data, grouping_key):
```
please improve the performance there are a lot of `for` statements

Squirrel18 replied:
Addressed. @andrey-canon

andrey-canon commented on proversity_reports_script/report_backend/time_spent_per_user_report.py:
```diff
@@ -0,0 +1,292 @@
+                    body_dict=time_spent_per_cohort_report_data,
+                    headers=time_spent_per_cohort_report_data[0].keys(),
+                    course_id=course_key,
+                    spreadsheet_id=self.spreadsheet_data.get('time_spent_sheet_id_{}'.format(course_key)),
```
you can get this inside the method

Squirrel18 replied:
Addressed. @andrey-canon

andrey-canon commented on proversity_reports_script/report_backend/time_spent_per_user_report.py:
```diff
@@ -0,0 +1,292 @@
+                self.create_csv_file(
+                    file_name='{}-cohorts'.format(file_name),
+                    body_dict=time_spent_per_cohort_report_data,
+                    headers=time_spent_per_cohort_report_data[0].keys(),
```
you can get this inside the method by using the body_dict

Squirrel18 replied:
Addressed. @andrey-canon

Squirrel18 committed:
Adding the time spent per user report.
Add the option to overwrite extra request parameters from command arguments.
Adding support for unknown command arguments.
